packages/lib/src/artifacts/ZWeb3.ts

```diff
@@ -140,8 +140,12 @@ export default class ZWeb3 {
     return ZWeb3.eth().net.getId();
   }
 
-  public static async getNetworkName(): Promise<string> {
+  public static async getNetworkName(providedNetworkId?: string | number): Promise<string | never> {
```

Comment:
Not sure I like doing this test here. We are abusing the fact that one of the first methods we call on the Web3 provider is `getNetworkName`. We should validate the network ID when the provider is set up, not in a random method.

Reply:
I agree, but I wouldn't change the `ZWeb3#initialize` implementation. WDYT if I add a new function to `ZWeb3`, something like `checkValidNetworkId` which will throw if the id is invalid? It will be called jsut after `ZWeb3#initialize`

Reply:
Sounds good!